packages/api/src/menus/services.ts

```diff
@@ -18,4 +18,39 @@ export const upsertMenuBatch = async (db: Drizzle, menuBatch: InsertMenu[]) =>
       date: sql`excluded.date`,
       price: sql`excluded.price`,
     },
-  });
\ No newline at end of file
+  });
+
+/* 
+ * Returns an object with the earliest
+ * and latest dates present in the menus table.
+*/
+export async function getPickableDates(db: Drizzle): Promise<DateList> {
+  const rows = await db
+    .selectDistinct({ date: menus.date })
+    .from(menus);
+  
+  if (!rows.length)
+    return null;
+
+  let uniqueDates: Date[] = [];
+  
+  rows.forEach((r) => {
+    const d = toLocalDate(r.date);
+    if (d)
+      uniqueDates.push(d);
+  });
+
+  return uniqueDates.sort((a, b) => a.getTime() - b.getTime());
+}
```

Comment:
We should move the sorting logic to the db as much as possible to ensure efficiency. I'd recommend an `orderBy()` following the select clause.

Additionally, we should leverage a mapping instead of pushing on to a new array.

```suggestion
export async function getPickableDates(db: Drizzle): Promise<DateList> {
  const rows = await db
    .selectDistinct({ date: menus.date })
    .from(menus)
    .orderBy(menus.date);
  
  if (!rows.length)
    return null;

  return rows
    .map((row) => toLocalDate(r.date))
    .filter((d) => d !== null);
}
```